// webflow-api/dist/serialization/types/LocalizedItem.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocalizedItem = void 0;
const core = __importStar(require("../../core"));
exports.LocalizedItem = core.serialization.object({
    cid: core.serialization.property("_cid", core.serialization.string()),
    id: core.serialization.property("_id", core.serialization.string()),
    locale: core.serialization.property("_locale", core.serialization.string()),
    draft: core.serialization.property("_draft", core.serialization.boolean()),
    archived: core.serialization.property("_archived", core.serialization.boolean()),
    name: core.serialization.string(),
    slug: core.serialization.string(),
    updatedOn: core.serialization.property("updated-on", core.serialization.date()),
    updatedBy: core.serialization.property("updated-by", core.serialization.string()),
    createdOn: core.serialization.property("created-on", core.serialization.date()),
    createdBy: core.serialization.property("created-by", core.serialization.string()),
    publishedOn: core.serialization.property("published-on", core.serialization.date().optional()),
    publishedBy: core.serialization.property("published-by", core.serialization.string().optional()),
});
